src/ExpressionEvaluator/Core/Source/ExpressionCompiler/MetadataUtilities.cs

```diff
@@ -95,8 +96,14 @@ internal static class MetadataUtilities
                 }
                 if (identitiesBuilder != null)
                 {
-                    var identity = metadata.MetadataReader.ReadAssemblyIdentityOrThrow();
+                    var reader = metadata.MetadataReader;
+                    var identity = reader.ReadAssemblyIdentityOrThrow();
                     identitiesBuilder.Add(identity);
+                    // If this assembly has no references, assume it is mscorlib.
+                    if (reader.AssemblyReferences.Count == 0)
+                    {
```

Comment:
I don't understand where or how this is run, but be aware that's valid to have an assembly without any references, and you do see this in the wild with satellites.